src/tokenizer.ts

```diff
@@ -1,8 +1,8 @@
-import type { Unlocated, Token, IdToken, Position } from './node-types';
+import type { Unlocated, Token, AttrToken, Position } from './node-types';
 
 const tagRegexp = /^<[/!]?(\w[\w-]*)(\s+[\w]+(\s*=\s*("[^"]*"|'[^']*'|[^><"'=``]+))?)*\s*>/;
 const commentRegexp = /^<!--[\w\W]*?-->/;
-const idRegexp = /^\[id="([\w-]+)"] /;
+const attrRegexp = /^\[ *[\w-]+ *= *"[^"\n]*" *(?:, *[\w-]+ *= *"[^"\n]*" *)*] /;
```

Comment:
More permissive string grammar so we can `JSON.parse` it below.

```suggestion
const attrRegexp = /^\[ *[\w-]+ *= *"(?:[^"\r\n\\]|\\["\\/bfnrt]|\\u[a-fA-F]{4})*" *(?:, *[\w-]+ *= *"(?:[^"\r\n\\]|\\["\\/bfnrt]|\\u[a-fA-F]{4})*" *)*] /;
```